app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java

```diff
@@ -131,7 +131,9 @@ private static Rect getBounds(@Nullable AccessibilityNodeInfo node, Rect display
         // Find the visible bounds of our first scrollable ancestor
         AccessibilityNodeInfo ancestor;
         int currentDepth = depth;
-        for (ancestor = node.getParent(); ancestor != null && ++currentDepth < MAX_DEPTH; ancestor = ancestor.getParent()) {
+        for (ancestor = node.getParent();
+             ancestor != null && ancestor != node && ++currentDepth < MAX_DEPTH;
```

Comment:
Codacy found an issue: [Use equals() to compare object references.](https://app.codacy.com/manual/dpgraham/appium-uiautomator2-server/pullRequest?prid=6083035)